kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt

```diff
@@ -490,6 +490,39 @@ fun Gen.Companion.choose(min: Long, max: Long): Gen<Long> {
   }
 }
 
+/**
+ * Returns a stream of values based on weights:
+ *
+ * <pre>Gen.choose(1 to 'A', 2 to 'B')</pre> will generate 'A' 33% of the time
+ * and 'B' 66% of the time.
+ *
+ * @throws IllegalArgumentException If any negative weight is given or any
+ * weight of zero is given.
+ */
+fun <T : Any> Gen.Companion.choose(a: Pair<Int, T>, b: Pair<Int, T>, vararg cs: Pair<Int, T>): Gen<T> {
+  val allPairs = listOf(a, b) + cs
+  val weights = allPairs.map { it.first }
+  require(weights.all { it >=0 }) { "Negative weights not allowed" }
+  require(weights.any { it > 0 }) { "At least one weight must be greater than zero"}
+  return object : Gen<T> {
+    // The algorithm for pick is a migration of
+    // the algorithm from Haskell QuickCheck
+    private tailrec fun pick(n: Int, l: Sequence<Pair<Int, T>>): T {
+      val (w, e) = l.first()
+      return if (n <= w) e
+      else pick(n - w, l.drop(1))
+    }
+    override fun constants(): Iterable<T> = emptyList()
```

Comment:
I'm wondering if we should keep the constants as a list of every possible outcome, ignoring weight (but I'm not sure that's even useful)

Reply:
@Kerooker I am not 100% clear on when the constants need to be present. Can you explain that a bit?